src/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.cs

```diff
@@ -717,40 +670,24 @@ public virtual void Emit(OpCode opcode, long arg)
         {
             EnsureCapacity(11);
             InternalEmit(opcode);
-            m_ILStream[m_length++] = (byte)arg;
-            m_ILStream[m_length++] = (byte)(arg >> 8);
-            m_ILStream[m_length++] = (byte)(arg >> 16);
-            m_ILStream[m_length++] = (byte)(arg >> 24);
-            m_ILStream[m_length++] = (byte)(arg >> 32);
-            m_ILStream[m_length++] = (byte)(arg >> 40);
-            m_ILStream[m_length++] = (byte)(arg >> 48);
-            m_ILStream[m_length++] = (byte)(arg >> 56);
+            BinaryPrimitives.WriteInt64LittleEndian(m_ILStream.AsSpan(m_length), arg);
+            m_length += 8;
         }
 
         public unsafe virtual void Emit(OpCode opcode, float arg)
         {
             EnsureCapacity(7);
             InternalEmit(opcode);
-            uint tempVal = *(uint*)&arg;
-            m_ILStream[m_length++] = (byte)tempVal;
-            m_ILStream[m_length++] = (byte)(tempVal >> 8);
-            m_ILStream[m_length++] = (byte)(tempVal >> 16);
-            m_ILStream[m_length++] = (byte)(tempVal >> 24);
+            BinaryPrimitives.WriteUInt32LittleEndian(m_ILStream.AsSpan(m_length), *(uint*)&arg);
```

Comment:
You can use `BitConverter.SingleToInt32Bits` here while you are on it to avoid the need for unsafe code.

Reply:
Good idea! Just to make sure, removing 'unsafe' from a methods signature isn't a breaking change for consumers, is it?

Reply:
https://github.com/dotnet/coreclr/pull/23833/commits/e55ce5573cc5e634cb91b0379cfa7e0b37f90063#diff-9251654972afbdb424e02431cb2b1e3cR677

Reply:
Its not a breaking change no. It's invisible to consumers of the API.

Reply:
Thanks!
Removed in https://github.com/dotnet/coreclr/pull/23833/commits/cfb3be31d0cba2e07500fb2922fed752c5bc3c4e